src/core/include/openvino/core/descriptor/tensor.hpp

```diff
@@ -107,6 +115,13 @@ class OPENVINO_API Tensor {
 
     void clone_from(const Tensor& old);
 
+private:
+    // set_element_type() is private func. To change Tensor element type please change the Parameter type.
+    void set_element_type(const element::Type& elemenet_type);
+
+    // set_tensor_type() is private func. To change Tensor type please change the Parameter type.
+    void set_tensor_type(const element::Type& element_type, const PartialShape& pshape);
```

Comment:
@ilya-lavrenov  you mean just to friend the methods which call the set_element_type() & set_tensor_type() not friend the whole class?

Reply:
If u only want to friend the call func, seems it can not implement, such as this func is a protect method.
https://github.com/openvinotoolkit/openvino/blob/bee44173ad50f57bfccc7a3712f7544a17de9e52/src/common/transformations/include/ov_ops/type_relaxed.hpp#L82-L92

Reply:
I meant that `set_tensor_type` can be made a friend of `ov::descriptor::Tensor` and other users can call this method.
But know, you've added all users of this method as friends.

Reply:
ok will do it